lib/profile/identity.rb

```diff
@@ -30,6 +30,13 @@ def self.find(name, cluster_type)
       all(cluster_type).find { |ident| ident.name == name }
     end
 
+    def self.set_conditional_dependencies(cluster_type, identity, dependencies)
+      metadata_path = File.join(cluster_type.base_path, "identities", identity, 'metadata.yaml')
+      metadata = YAML.load_file(metadata_path)
+      metadata['conditional_dependencies'] = dependencies
+      File.write(metadata_path, metadata.to_yaml)
+    end
```

Comment:
I don't think directly updating the identity metadata file is a good idea. The files belonging to `flight-profile-types` were not designed to be updated in-place. For example, if the user updates their `flight-profile-types` package, any changes that have been made to the `conditional_dependencies` field will be overwritten.

Is there a way you can change this so that, when Profile resolves dependencies for a given node, it checks the `dependencies` key of the type's `metadata.yaml` file instead of the identity's `metadata.yaml` file?